homeassistant/components/intellifire/binary_sensor.py

```diff
@@ -58,6 +60,92 @@ class IntellifireBinarySensorEntityDescription(
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.error_flame,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    IntellifireBinarySensorEntityDescription(
+        key="error_fan_delay",
+        name="Fan Delay Error",
+        icon="mdi:fan-alert",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.error_fan_delay,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    IntellifireBinarySensorEntityDescription(
+        key="error_maintenance",
+        name="Maintenance Error",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.error_maintenance,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    IntellifireBinarySensorEntityDescription(
+        key="error_disabled",
+        name="Disabled Error",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.error_disabled,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    IntellifireBinarySensorEntityDescription(
+        key="error_fan",
+        name="Fan Error",
+        icon="mdi:fan-alert",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.error_fan,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    IntellifireBinarySensorEntityDescription(
+        key="error_lights",
+        name="Lights Error",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.error_lights,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    IntellifireBinarySensorEntityDescription(
+        key="error_accessory",
+        name="Accessory Error",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.error_accessory,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    IntellifireBinarySensorEntityDescription(
+        key="error_soft_lock_out",
+        name="Soft Lock Out Error",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.error_soft_lock_out,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    IntellifireBinarySensorEntityDescription(
+        key="error_ecm_offline",
+        name="ECM Offline Error",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.error_ecm_offline,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    IntellifireBinarySensorEntityDescription(
+        key="error_offline",
+        name="Offline Error",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.error_offline,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    IntellifireBinarySensorEntityDescription(
+        key="has_errors",
+        name="Error(s)",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.errors != [],
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
```

Comment:
```suggestion
```

Generally we don't want duplicates data in the state machine.

Reply:
OOPS -> I should have just hit the commit suggestion button!  I was reading from my phone last night... oh well I pushed the change.